### PR TITLE
subunit, python310Packages.subunit: 1.4.0 -> 1.4.2

### PR DESCRIPTION
**Pull request by tjni**
###### Description of changes

https://github.com/testing-cabal/subunit/blob/master/NEWS

Related to https://github.com/NixOS/nixpkgs/pull/154650 and https://github.com/NixOS/nixpkgs/pull/154264.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/subunit/default.nix [hunk 1]
```diff
@@ -10,29 +10,43 @@
...
 }:
 
 buildPythonPackage {
   inherit (subunit) name src meta;
+  format = "pyproject";
+
+  disabled = pythonOlder "3.6";
+
+  postPatch = ''
+    sed -i 's/version=VERSION/version="${subunit.version}"/' setup.py
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subunit/default.nix:25]:
```suggestion
    substituteInPlace setup.py \
      --replace "version=VERSION" 'version="${subunit.version}"'
```

File: pkgs/development/python-modules/subunit/default.nix [hunk 2]
```diff
@@ -10,29 +10,43 @@
...
-  nativeBuildInputs = [ pkg-config ];
   buildInputs = [ check cppunit ];
   propagatedBuildInputs = [ testtools ];
 
-  checkInputs = [ testscenarios hypothesis fixtures pytest unittest2 ];
+  checkInputs = [
+    testscenarios
+    hypothesis
+    fixtures
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subunit/default.nix:40]:
```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/subunit/default.nix [hunk 3]
```diff
@@ -10,29 +10,43 @@
...
   checkPhase = ''
+    runHook preCheck
+
     pytest python/subunit \
       --ignore=python/subunit/tests/test_{output_filter,test_protocol{,2}}.py
-  '';
 
-  postPatch = ''
-    sed -i 's/version=VERSION/version="${subunit.version}"/' setup.py
+    runHook postCheck
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subunit/default.nix:50]: Please adopt to pytestCheckHook

File: pkgs/development/python-modules/subunit/default.nix (hunk 1 above)
Comment (tjni) [reply pkgs/development/python-modules/subunit/default.nix]: Nice, done!

File: pkgs/development/python-modules/subunit/default.nix (hunk 3 above)
Comment (tjni) [reply pkgs/development/python-modules/subunit/default.nix]: I updated this to use pytestCheckHook.

**Pull request merged**